articles/Frontend/Components/calculator.md

```diff
@@ -0,0 +1,867 @@
+loaded by `mathsEvaluator` and then linked to the parser.
+- `tokenizer`: A wrapper factory that initializes and gives access to the lexer
+used to tokenize the expressions. This tokenizer is used to manipulate the
+expression and properly display it.
+- `moo`: The third-party tokenizer that is reponsible to split the expression
+into atomic elements. It is not directly used by the calculator, as it is
+wrapped by the `tokenizer` wrapper factory. 
+- `tokens`: A helper module that ease the processing of tokens (atomic elements)
+extracted from expressions. It provides functions to qualify tokens by
+categories, transform them into string values, or render them in HTML. 
+- `terms`: The list of terms an expression can use. This is used to feed the
+tokenizer's engine and also to get the meaning of each token. 
+- `labels`: The labels corresponding to the terms used in expressions, and the
+labels usable for the display. There are directly linked by the list of terms,
+as well as by the keyboard layouts.
+- `areaBroker`: The component responsible to manage the display layout, giving
+access to the placeholders.
+- `plugin`: The plugins abstraction on which each calculator plugin should rely
+to be compatible with the calculator.
+
+Events model is the core concept allowing to communicate with the calculator
+`board`, but a particular mechanism has also been built in to interact with the
+expression: the commands layer.
+
+### Commands layer
+The commands layer is built on top of the events model, as it relies on events
+to trigger each command. A strong difference though, with standard events,
+should be mentioned: commands have to be registered, otherwise an error event
+will be emitted when a client will try to use them.
+
+Commands are the recommended way to add behavior on the calculator, as they
+provide a particular meaning. Each plugins can then expose their particular
+commands, or simply link to existing ones. The calculator `board` is also
+defining standard commands, like the general purpose cleaning commands or the
+execute computation one. The list of known commands will be detailed later in
+this doc.  
+
+### API
+The calculator `board` is the main component, exposing the essential API to
+manage the calculator and process to computations. However, by default nothing
+is displayed, as it focuses on the engine layer. Display and other fancy stuff
+are delegated to plugins, which are using the events and the API to allow the
+user to interact with the component. The `board` is an abstract component.
+
+#### calculator board factory
+The entry point to this calculator `board` is its factory function: 
+`calculatorBoardFactory($container, plugins, config)`, that returns an instance
+of the component.
+
+- `$container`: The first parameter is the DOM container in which the component
+will be rendered. This is mandatory since the area broker component requires
```

Comment:
$ indicates jQuery, it should therfor the DOM container, repesented by a jQuery Object,